### PR TITLE
State machine clean

### PR DESCRIPTION
**Pull request by Fberbari**
With this PR, we've got a working state machine and its accompanied unit tests. Although some of the modules inside it (like PID) need some work, the state machine flow and data hand off functions correctly.

Changes that were made that don't seem obvious to the purpose of this commit are the following:

 Autopilot/CMakeLists.txt : Since so many new unit tests came in, I had to create multiple unit test executables (rather than 1 massive one). That way, different test suites could target exactly what they're interested in.

Autopilot/Tools/build.bash : Supports the above Cmake change.

Movement of the SensorFusion and OutputMixing unit test files: Better directory structure of the Test folderwas added to accommodate the above change

### CODE REVIEW

**Review by sahil-kale — APPROVED** [review]
Didn't see anything out of the ordinary :)

File: Autopilot/AttitudeManager/attitudeStateClasses.hpp [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        void exit(attitudeManager* attitudeMgr) {(void) attitudeMgr;}
+        static attitudeState& getInstance();
+    private:
+        FatalFailureMode() {}
+        FatalFailureMode(const FatalFailureMode& other);
+        FatalFailureMode& operator =(const FatalFailureMode& other);
+};
+
+
+//Reminder: add failed state mode?
```
Comment (heyitslu) [inline Autopilot/AttitudeManager/attitudeStateClasses.hpp:127]: Failed state mode was added already haha

**Review by heyitslu — APPROVED** [review]
Looks good to me!

File: Autopilot/AttitudeManager/attitudeStateClasses.hpp (hunk 1 above)
Comment (Fberbari) [reply Autopilot/AttitudeManager/attitudeStateClasses.hpp:127]: good eye lmao, will fix next time this file is touched

**Pull request merged**